libfreerdp/gdi/gfx.c

```diff
@@ -929,7 +936,8 @@ static UINT gdi_SurfaceCommand_Progressive(rdpGdi* gdi, RdpgfxClientContext* con
 	WINPR_ASSERT(gdi);
 	WINPR_ASSERT(context);
 	WINPR_ASSERT(cmd);
-	surface = (gdiGfxSurface*)context->GetSurfaceData(context, cmd->surfaceId);
+	const UINT16 surfaceId = (UINT16)MIN(UINT16_MAX, cmd->surfaceId);
```

Comment:
I don't understand why the surfaceId is capped to UINT16...

Reply:
because it should be a `UINT16` on the wire (maybe we should change the datatype)